src/Mesh/Topologies.jl

```diff
@@ -1052,6 +1052,74 @@ function grid_stretching_1d(coord_min, coord_max, Ne, stretching_type, attractor
     return range_stretched
     
 end
-#}}}
 
+
+
+
+function Generate_grid(::Val{3},DDims,ts::Float64,tf::Int64,mpicomm::MPI.Comm,O::Int64,DomainSize,periodicity,Boundary)
```

Comment:
Rather than creating new functions, it might be better to just add new methods to `StackedBrickTopology`: e.g.

```julia

function approxstep_range(a,b,step)
    range(a, stop=b, length=round(Int,abs(b-a)/step))
end
    
function StackedBrickTopology(mpicomm, domainranges::NTuple{N,Tuple{T,T}}, deltas::NTuple{N,T};
                       polynomialorder,
                       boundary=ntuple(j->(1,1), N),
                       periodicity=ntuple(j->false, N)) where {N,T}
   
   brickranges = map(((a,b), delta) -> approxstep_range(a, b, delta*polyorder), domainranges, deltas)
   StackedBrickTopology(mpicomm, brickranges, boundary=boundary, periodicity=periodicity)
end
```

Reply:
`StackedBrickTopology` already exists as a function: you would just be adding additional methods to it.

Reply:
`StackedBrickTopology` already exists: you would just be adding additional methods to it. This creates a new function.

StackedBrickTopology already exists as a function: you would just be adding additional methods to it.